docs/rfc/rfc-026-p2p-bad-peers-checktx.md

```diff
@@ -0,0 +1,432 @@
+is not valid. Thus, the main role of `CheckTx` is to, as early as possible, prevent invalid transactions
+from entering the mempool. 
+
+Note that there are valid scenarios in which a transaction will not
+pass this check (including nodes getting different transactions at different times, meaning some
+of them might be obsolete at the time of the check; state changes upon block execution etc.). 
+However, Tendermint users observed that
+there are transactions that can never have been or will never be valid. They thus propose
+to introduce a special response code for `CheckTx` to indicate this behaviour, and ban the peers
+who gossip such transactions. Additionally, users expressed a need for banning peers who
+repeatedly send transactions failing `CheckTx`. 
+
+The main goal of this document is to analyse the cases where peers could be banned when they send 
+transactions failing `CheckTx`, and provide the exact conditions that a peer and transaction have
+to satisfy in order to mark the peer as bad.
+
+This document will also include a proposal for implementing these changes within the mempool, including
+potential changes to the existing mempool logic and implementation.
+
+
+## Background
+
+This work was triggered by issue [#7918](https://github.com/tendermint/tendermint/issues/7918) and a related
+discussion in [#2185](https://github.com/tendermint/tendermint/issues/2185). Additionally,
+there was a [proposal](https://github.com/tendermint/tendermint/issues/6523) 
+to disconnect from peers after they send us transactions that constantly fail `CheckTx`.  While 
+the actual implementation of an additional response code for `CheckTx` is straight forward there 
+are certain correctness aspects to consider. The questions to answer, along with identified risks will be outlined in 
+the discussion. 
+
+### Existing issues and concerns
+
+Before diving into the details, we collected a set of issues opened by various users, arguing for 
+this behaviour and explaining their needs. 
+
+- Celestia: [blacklisting peers that repeatedly send bad tx](https://github.com/celestiaorg/celestia-core/issues/867)
+  and investigating how [Tendermint treats large Txs](https://github.com/celestiaorg/celestia-core/issues/243)
+- BigChainDb: [Handling proposers who propose bad blocks](https://github.com/bigchaindb/BEPs/issues/84)
+- IBC-go: Simulate transactions in the mempool and detect whether they could ever have been valid to 
+  [avoid overloading the p2p/mempool system](https://github.com/cosmos/ibc-go/issues/853#issuecomment-1032211020)
+ (ToDo check with Adi what is this about, what are valid reasons this can happen and when should we disconnect)
+  (relevant code [here](https://github.com/cosmos/ibc-go/blob/a0e59b8e7a2e1305b7b168962e20516ca8c98fad/modules/core/ante/ante.go#L23)) 
+- IBC relayers: Nodes allow transactions with a wrong `minGas` transaction and gossip them, and other nodes keep rejecting them. (Problem seen with relayers)
```

Comment:
For completeness, this is the exact config name and link to it:

https://github.com/cosmos/gaia/blob/5db8fcc9a229730f5115bed82d0f85b6db7184b4/contrib/testnets/test_platform/templates/app.toml#L8-L11